packages/toolkit/src/lib/react-query-service/mgmt/useIntegrationConnection.ts

```diff
@@ -0,0 +1,45 @@
+import { useQuery } from "@tanstack/react-query";
+import { Nullable, ResourceView } from "instill-sdk";
+
+import { getInstillAPIClient } from "../../vdp-sdk";
+
+export function useIntegrationConnection({
+  enabled,
+  retry,
+  view = "VIEW_BASIC",
+  connectionId,
+  namespaceId,
+  accessToken,
+}: {
+  accessToken?: string;
+  enabled: boolean;
+  retry?: false | number;
+  view: ResourceView;
+  connectionId: Nullable<string>;
+  namespaceId: Nullable<string>;
+}) {
+  return useQuery({
+    queryKey: ["integration-connection", connectionId, namespaceId, view],
+    queryFn: async () => {
+      if (!connectionId) {
+        return Promise.reject(new Error("integrationId not provided"));
+      }
+
+      if (!namespaceId) {
+        return Promise.reject(new Error("namespaceId not provided"));
+      }
+
```

Comment:
@orangecoloured We need to check accessToken here since this is a query for private resource